spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/config/SetStatusProperties.java

```diff
@@ -0,0 +1,30 @@
+package org.springframework.cloud.gateway.config;
```

Comment:
I'd prefer to move these directly to `SetStatusGatewayFilterFactory.java`. We've done it in other filters.

Reply:
As I can see in other filters inner class configs are about configuration per route.
I found LoadBalancerProperties and do this one in the same way as far as this configuration applicable to all routes with the configured filter.

Reply:
See [RemoveHopByHopHeadersFilter](https://github.com/spring-cloud/spring-cloud-gateway/blob/7e4f44a46e6251f71d9e04b1ab5c205d3f4327c3/spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/filter/headers/RemoveHopByHopHeadersFilter.java#L28), [RequestRateLimiterGatewayFilterFactory](https://github.com/spring-cloud/spring-cloud-gateway/blob/7e4f44a46e6251f71d9e04b1ab5c205d3f4327c3/spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/filter/factory/RequestRateLimiterGatewayFilterFactory.java#L35), and [XForwardedHeadersFilter](https://github.com/spring-cloud/spring-cloud-gateway/blob/7e4f44a46e6251f71d9e04b1ab5c205d3f4327c3/spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/filter/headers/XForwardedHeadersFilter.java#L33)

Reply:
Personally, I do like more the implementation with a separate properties file.
but it seems to be more consistent to use @ConfigurationProperties on top of the Factory.
And according to the below comment will leave the originalStatusHeaderName only field

Reply:
Let's move it please.

Reply:
done